src/lib/pipe_lib/strict_pipe.ml

```diff
@@ -54,6 +54,36 @@ module Reader = struct
     assert_not_read reader ;
     reader.has_reader <- true ;
     wrap_reader (Pipe.filter_map reader.reader ~f)
+
+  let iter_sync ?consumer ?continue_on_error reader ~f =
+    iter ?consumer ?continue_on_error reader ~f:(fun x -> f x ; Deferred.unit)
+
+  module Merge = struct
+    let iter readers ~f =
```

Comment:
Can this use `Pipe.interleave`?

Reply:
`Pipe.interleave` does not prioritize the readers in the order they are provided. The goal of this function is to merge pipes while maintaining read priority in the order of the list provided.